### PR TITLE
requirement must use official version in order to publish them on pypi

### PR DESCRIPTION
**Pull request by sebastienbeau**
@hparfr to discuss
We can no publish to pypi if we depend of a fork
@PierrickBrun

### CODE REVIEW

**Comment by hparfr** [comment]
Yes but we can't use branch without the fork.

**Comment by rvalyi** [comment]
two possible workarounds:

1. publish our own forked copier package under a different name. Cons: would force us to maintain it.
2. monkey patch copier. Cons: monkey patching should be used wisely...

I didn't think further what would be the best...

**Comment by rvalyi** [comment]
@sebastienbeau I had to do this trivial fix to get it to work https://github.com/akretion/docky/pull/154/commits/a714674f92ec61ef047dbbd62dbabf8e57a77bca

**Review by rvalyi — APPROVED** [review]
got it working ( https://github.com/akretion/docky/issues/155#issuecomment-1312493016 ) and yes I think any feature that prevent basic things such as pypi distribution should be discussed wisely before inclusion. Ideally we have something basic and small we can all agree with and extra stuff is added in extra layers.

**Review by PierrickBrun — APPROVED** [review]

**Comment by bealdav** [comment]
OK for me to not use a fork of copier. I really prefer to know that we use explicitely copier and not encapsulate it

**Comment by sebastienbeau** [comment]
@hparfr let's merge it ?

**Review by bealdav — APPROVED** [review]

File: docky/cmd/init.py [hunk 1]
```diff
@@ -3,21 +3,28 @@
...
-    
-    def main(self, *args, **kwargs):
-        GenerateProject().generate(url=self.template_url, branch=self.template_branch)
+    """Initalize the file '.env' of an existing project"""
+    def _uid(self):
+        return id['-u']().replace('\n', '')
+
+    def main(self):
+        if not local.path(".env").is_file():
+            env = prompt("Choose the env to init (dev, prod...)?", default="dev")
```
Comment (hparfr) [inline docky/cmd/init.py]: this should be done in copier template instead

**Review by hparfr — CHANGES_REQUESTED** [review]
There is only to acceptable choices:
- [ ] docky init + copier (forked or not is another debate)
- [x] no docky init at all.

This simplistic implementation of .env is not good enough

**Comment by bealdav** [comment]
yes maybe no docky init is an option if .env created by copier !? @sebastienbeau

File: docky/cmd/init.py (hunk 1 above)
Comment (PierrickBrun) [reply docky/cmd/init.py]: About the .env file, I'm not sure if copier is the way to do it (maybe init it but without the real values)

IMO copier shouldn't care about the environment and is meant to be run in a dev env to initalize/update the project (and push it in git to deploy it in the other environments)

the .env file is ignored by git and should then use another path. Maybe docky is the way to do it ? (but with an adequate naming)

It feels weird to have to run copier update both on dev and prod.

**Comment by sebastienbeau** [comment]
Copier should not create the ".env" file as this file is not commited.
It's not the logic of copier, copier is here to create, then update the versioned file.
Moreover you want to generate an env file without updating your project, so copier is not the right tools for doing it.

So we have the following possibility for the .env
- no generation (you do a copy paste from other project, and we need to document it)
- having an example of .env called .env-example in the copier template
- adding a check in docky run and if the file is missing propose to create one with default value (that can be updated manually)

What do you think ?

**Comment by bealdav** [comment]
\+ 1 for .env-sample with instruction inside

**Comment by PierrickBrun** [comment]
.env-sample seems good to me

It would be great if docky could refuse to start if there is no .env file also

**Comment by sebastienbeau** [comment]
Docky init removed, let's merge it

**Comment by hparfr** [comment]
please update the readme also ;-)

**Comment by sebastienbeau** [comment]
@hparfr readme is ok for you ?

**Review by hparfr — APPROVED** [review]

**Pull request merged**